src/main/java/TaskList.java

```diff
@@ -0,0 +1,194 @@
+     */
+    public void addTodoTask(String description) throws DukeException {
+        if (description.isBlank()) {
+            //for the case where user keys in a space after the command
+            throw new DukeException("     ☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        Todo newTodoTask = new Todo(description);
+        tasks.add(newTodoTask);
+        ui.printAddedTask(newTodoTask, tasks.size());
+    }
+
+    /**
+     * Formats the date string from "yyyy-mm-dd" to a format of "d MMM yyyy".
+     *
+     * @param dateString the string that is to be formatted and in the form of "yyyy-mm-dd".
+     * @return the newly formatted string in "d MMM yyyy" format.
+     */
+    public String formatDate(String dateString) {
+        int firstHyphenIndex = dateString.indexOf("-");
+        int dateStartIndex = firstHyphenIndex - 4;
+        int dataEndIndex = firstHyphenIndex + 6;
+        String extractedDate = dateString.substring(dateStartIndex, dataEndIndex);
+        LocalDate date = LocalDate.parse(extractedDate);
+        String formattedDate =  date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        return dateString.replace(extractedDate, formattedDate);
+    }
+
+    /**
+     * Adds a task of type Deadline to the task list given the correct format, formats the date given in the deadline
+     * if given in a certain format. If addition succeeds, proceeds to print the Deadline task that is added
+     * to the task list.
+     *
+     * @param description the details of the Deadline task to be added.
+     * @throws DukeException if the description of the Deadline task is empty.
+     */
+    public void addDeadlineTask(String description) throws DukeException {
+        if (description.isBlank()) {
+            //for the case where user keys in a space after the command
+            throw new DukeException("     ☹ OOPS!!! The description of a deadline cannot be empty.");
+        }
+        int firstSlashIndex = description.indexOf("/by");
+        int detailsEndIndex = firstSlashIndex - 1;
+        String details = description.substring(0, detailsEndIndex);
+        int byStartIndex = firstSlashIndex + 4;
+        String by = description.substring(byStartIndex);
+        if (by.contains("-")) { //case where user keys in a date
+            by = formatDate(by);
+        }
+        Deadline newDeadlineTask = new Deadline(details, by);
+        tasks.add(newDeadlineTask);
+        ui.printAddedTask(newDeadlineTask, tasks.size());
+    }
+
+    /**
+     * Adds a task of type Event to the task list given the correct format.
+     * If addition succeeds, proceeds to print the Event task that is added to the task list.
+     *
+     * @param description the details of the Event task to be added.
+     * @throws DukeException if the description of the Event task is empty.
+     */
+    public void addEventTask(String description) throws DukeException {
+        if (description.isBlank()) {
```

Comment:
Good usage of the concept of 'keeping the happy path prominent.' [See here]( https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent)